core/drivers/tpm2/tpm2_ptp_fifo.c

```diff
@@ -369,10 +372,8 @@ enum tpm2_result tpm2_fifo_send(struct tpm2_chip *chip, uint8_t *buf,
 		return TPM2_OK;
 	}
 
-	/* This will result in current command getting cancelled */
-	tpm2_fifo_set_status(chip, TPM2_STS_COMMAND_READY);
-	/* We are here because some error occurred, so relinquish locality */
-	tpm2_fifo_relinquish_locality(chip);
+	/* Cancel command and relinquish locality */
+	ret = tpm2_fifo_end(chip);
```

Comment:
sorry I confused you here.
Sorry not overwrite returned `ret` value.
I would propose:
```c
	/* Cancel command and relinquish locality */
	if (tpm2_fifo_end(chip))
		return TPM2_ERR_GENERIC;

	return ret;
```
Ditto at line 480